### ISSUE TITLE
Language server does not start without Eclipse project files

### ISSUE THREAD

**Issue by RolKau** [redhat-developer/vscode-java#2877]
From [this blog entry](https://devblogs.microsoft.com/java/say-goodbye-to-project-files-in-1-1-0/) it seems that the old Eclipse project files; .project and .classpath, would no longer be neccessary. However, this is not the case: The files are no longer auto-generated, but if they are not present, the project won't open correctly.

##### Environment
- Operating System: Ubuntu Linux 20.04
- JDK version:  OpenJDK 17.0.5
- Visual Studio Code version: 1.74.3
- Java extension version: 1.13.0

##### Steps To Reproduce
1. Unzip the attached project.zip into a new directory
2. Open the project in VSCode through the project file foo.code-workspace
3. Observe that after the View Projects popup has appeared, there is a project in "Java Projects" called "foo", and that you can go into the Program.java file, right-click on the word "Program" after "class" on the first line, and select Find All References from the menu; three results will appear in the References panel.
4. Type Ctrl-Shift-P, Java: Open Java Language Server Log File; this will be a file underneath the directory ~/.config/Code/User/workspaceStorage/xxx; quit VSCode and remove this directory xxx.
5. Then delete the .project and .classpath files in the project directory.
6. Open the project again in the same way as in step 2. However, the Java Project panel will now contain no projects, and if you open the Program.java file and right-click on the "Program" identifier after the "class" keyword and select Find All References you get no results.
7. Quit VSCode and restore the .project and .classpath files.
8. Repeat step 2 and 3; the project now works again after the files were restored.

##### Current Result
See step 6 above; the language server appears broken.

##### Expected Result
See step 3 above; the language server should work.

##### Additional Informations
[project.zip](https://github.com/redhat-developer/vscode-java/files/10410003/project.zip)
[client.log](https://github.com/redhat-developer/vscode-java/files/10410475/client.log)

If you have a multi-module Maven project which was imported in an old version of VSCode, and had .project and .classpath generated for the existing modules at the time, and you now add a new module which will not have them generated,the project looks partially broken.

**Comment by jdneo** [comment]
It's because the sample project has some issue during importing.

If you delete the `.classpath` & `.project` and then try to import it as a maven project in Eclipse, you can see that the `.classpath` file is not generated and features are not working:

![image](https://user-images.githubusercontent.com/6193897/212617744-a3717260-58f8-4292-ba9e-47d65d80abb0.png)

One possible reason is that the m2e does not support project that marks the base path as the source root.

**Comment by RolKau** [comment]
> One possible reason is that the m2e does not support project that marks the base path as the source root

That is not the reason, because the original project where I discovered this problem did not have that property; it was only to make the directory structure in the sample simpler. Also, note that the project was successfully imported in step 3 in the issue description, before I deleted `.project` and `.classpath`.

If you move the file `Program.java` into a sub-directory `src/main/java` and delete the `<sourceDirectory>` tag in `pom.xml`, it exhibits exactly the same behaviour.
